### ISSUE TITLE
[BUG] Opening console causes crash if console cleared before opening for the first time

### ISSUE THREAD

**Issue by 72hr** [ZDoom/gzdoom#2047]
### GZDoom version

GZDoom 4.10.0

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Windows 10

### Please describe your specific OS version

Windows 10 Pro

### Relevant hardware info

Ryzen 3600, RTX 3060 Ti

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

To avoid having game messages appear over the stats in the upper left corner of the map, I added this macro to my autoexec:

```
alias toggle-map "clear; togglemap"
bind tab toggle-map
```

I started getting crashes when opening the console and I've found that it only happens if I've opened the map before opening the console for the first time.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. bind `clear` to a key
2. launch gzdoom, start a game
3. press that key
4. open the console

### Your configuration

this occurs with a default config with the following line added to the bindings:

```ini
 N=clear
```

**Comment by 50hr** [comment]
Right before the crash, c_console.cpp:650 `mBrokenLines` is empty and `FBrokenLines *blines = conbuffer->GetLines();` is a null pointer, and then `FBrokenLines *printline = blines + consolelines - 1 - RowAdjust;` is 0xfffffffffffffff0, so it enters the for loop and crashes inside, I think on `p->Text`? Would a null check on `blines` be an OK solution here? Like 
```cpp
int bottomline = ConBottom / textScale - CurrentConsoleFont->GetHeight() * 2 - 4;
FBrokenLines *blines = conbuffer->GetLines();
if (blines != NULL)
{
    FBrokenLines* printline = blines + consolelines - 1 - RowAdjust;
    for (FBrokenLines* p = printline; p >= blines && lines > 0; p--, lines--)
    {
    // ...
```

**Comment by coelckers** [comment]
fixed in 30c2524742ae3ef8da3c8ee73fab9ce6b562cf1d
